digdag-tests/src/test/java/acceptance/RequireIT.java

```diff
@@ -185,6 +189,55 @@ public void testIgnoreProjectIdParam()
         assertThat(success, is(true)); // --param project_id=-1 is ignored.
     }
 
+
+    /**
+     * Test for project_id and project_name parameter
+     * @throws Exception
+     */
+    @Test
+    public void testRequireToAnotherProject()
+            throws Exception
+    {
+        final String childProjectName = "child_another";
+
+        // Push child project
+        Path childProjectDir = folder.getRoot().toPath().resolve("another_foobar");
+        Files.createDirectories(childProjectDir);
+        copyResource("acceptance/require/child_another_project.dig", childProjectDir);
+        CommandStatus pushChildStatus = main("push",
+                "--project", childProjectDir.toString(),
+                childProjectName,
+                "-c", config.toString(),
+                "-e", server.endpoint());
+        assertThat(pushChildStatus.errUtf8(), pushChildStatus.code(), is(0));
+
+        // extract child project id
+        Matcher m = Pattern.compile(".*\\s+id:\\s+(\\d+).*").matcher(pushChildStatus.outUtf8());
+        assertThat(m.find(), is(true));
+        String childProjectId = m.group(1);
+
+        // Push parent project
+        Files.write(projectDir.resolve("parent_another_project.dig"), asList(Resources.toString(
+                Resources.getResource("acceptance/require/parent_another_project.dig"), UTF_8)
+                .replace("__CHILD_PROJECT_ID__", childProjectId)
+                .replace("__CHILD_PROJECT_NAME__", childProjectName)));
+        CommandStatus pushParentStatus = main("push",
+                "--project", projectDir.toString(),
+                "parent_another",
+                "-c", config.toString(),
+                "-e", server.endpoint());
+        assertThat(pushParentStatus.errUtf8(), pushParentStatus.code(), is(0));
+
+        CommandStatus startStatus = main("start",
+                "-c", config.toString(),
+                "-e", server.endpoint(),
+                "parent_another", "parent_another_project",
+                "--session", "now"
+        );
+        assertThat(startStatus.errUtf8(), startStatus.code(), is(0));
```

Comment:
Does this test check if the child project workflow has run as expected?

Reply:
Oh, it is not enough.
Need to check the result of attempt. (must be success).
I will add check code.

Reply:
Thanks!

Reply:
To make test robust, I separated the project_id and project_name test and check attempt result both parent and its child.

- testRequireToAnotherProjectById()
- testRequireToAnotherProjectByName()